src/runtime_src/core/common/pskernel_parse.cpp

```diff
@@ -0,0 +1,230 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#define XRT_CORE_COMMON_SOURCE
+#include <regex>
+#include "pskernel_parse.h"
+
+
+namespace xrt_core { namespace pskernel {
+
+// Function to match function name
+// and extract formal parameters
+std::vector<xrt_core::pskernel::kernel_argument>
+extract_args (Dwarf_Die *die)
+{
+  Dwarf_Die child;
+  Dwarf_Attribute attr_mem;
+  Dwarf_Die type_mem;
+  Dwarf_Die *type = NULL;
+  Dwarf_Word var_size;
+  const char *var_name;
+  std::vector<xrt_core::pskernel::kernel_argument> return_args;
+
+  // Mapping for FFI types
+  static const std::map<std::pair<Dwarf_Word, Dwarf_Word>, ffi_type*> typeTable = {
+    std::make_pair(std::make_pair(DW_ATE_unsigned_char, 1), &ffi_type_uint8),
+    std::make_pair(std::make_pair(DW_ATE_signed_char, 1), &ffi_type_sint8),
+    std::make_pair(std::make_pair(DW_ATE_unsigned, 2), &ffi_type_uint16),
+    std::make_pair(std::make_pair(DW_ATE_signed, 2), &ffi_type_sint16),
+    std::make_pair(std::make_pair(DW_ATE_unsigned, 4), &ffi_type_uint32),
+    std::make_pair(std::make_pair(DW_ATE_signed, 4), &ffi_type_sint32),
+    std::make_pair(std::make_pair(DW_ATE_unsigned, 8), &ffi_type_uint64),
+    std::make_pair(std::make_pair(DW_ATE_signed, 8), &ffi_type_sint64),
+    std::make_pair(std::make_pair(DW_ATE_float, 4), &ffi_type_float),
+    std::make_pair(std::make_pair(DW_ATE_float, 8), &ffi_type_double)
+  };
+  
+  int offset = 4;
+  int index = 0;
+
+  if (dwarf_child (die, &child) == 0)
+    do
+      switch (dwarf_tag (&child)){
+	case DW_TAG_formal_parameter:
+	  // Extract parameter name and type
+	  xrt_core::pskernel::kernel_argument arg;
+	  var_name = dwarf_diename(&child);
+	  arg.name = var_name;
+	  type = dwarf_formref_die(dwarf_attr(&child, DW_AT_type, &attr_mem),&type_mem);
+	  while(dwarf_tag(type) == DW_TAG_typedef) {
+	    type = dwarf_formref_die(dwarf_attr(type, DW_AT_type, &attr_mem),&type_mem);
+	  }
+	  if(dwarf_tag(type) == DW_TAG_base_type) {
+	    Dwarf_Attribute encoding;
+	    Dwarf_Word enctype = 0;
+	    if (dwarf_attr (type, DW_AT_encoding, &encoding) == NULL
+		|| dwarf_formudata (&encoding, &enctype) != 0)
+	      throw std::runtime_error("base type without encoding");
+	    
+	    Dwarf_Attribute bsize;
+	    Dwarf_Word bits;
+	    if (dwarf_attr (type, DW_AT_byte_size, &bsize) != NULL
+		&& dwarf_formudata (&bsize, &bits) == 0) {
+	      arg.size = bits;
+	    } else if (dwarf_attr (type, DW_AT_bit_size, &bsize) == NULL
+		     || dwarf_formudata (&bsize, &bits) != 0)
+	      throw std::runtime_error("base type without byte or bit size");
+
+	    auto result = typeTable.find(std::make_pair(enctype,bits));
```

Comment:
```suggestion
	    auto result = typeTable.find({ enctype, bits});
```
and fix the space/tab mixture all araound the neighborhood.